### PR TITLE
Add backend batch operation for update/delete template versions

### PR DESCRIPTION
**Pull request by gracez72**
Ticket: [TASK-33741](https://microsoftgarage.visualstudio.com/Intern%20GitHub/_sprints/taskboard/Adaptive%20Cards/Intern%20GitHub/Y20-VW/10?workitem=33741)

**Issue**
Modifying the state of multiple template versions causes concurrency issues when writing/reading to the Mongo database. Adding batch endpoints for deleting, publishing, and unpublishing multiple template versions lessens the load on the database to lock until updates are done. 

**New Endpoints**
- DELETE /template/{templateId}/batch
```
{
  "versions": ["1.0", "1.1"]
}
```
- POST /template/{templateId}/batch
```
{
  "templates" : [
      {
          "version": "1.0", 
          "state": "deprecated"
      } 
  ]
}
```

**Changes to Review**
- `swagger.yaml` - definitions for the client SDK, this definition autogenerates the changes under `adaptivecards-templating-client/typescript`
- `TemplateServiceClient.ts`: backend endpoint logic
- Files under `client/src`: Changed functions under version modal to use batch endpoints

### CODE REVIEW

File: private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts [hunk 1]
```diff
@@ -455,6 +455,65 @@ export class TemplateServiceClient {
...
+    let userResponse = await this._getUser(authId);
+
+    if (!userResponse.success) {
+      return { success: false, errorMessage: userResponse.errorMessage };
+    }
+
+    // Check if version already exists
+    let response = await this.getTemplates(token, templateId);
+    if (!response.success || !response.result || response.result.length === 0) {
+      return { success: false };
```
Comment (turalf) [inline private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts]: you can add errorMessage here as well I guess, letting caller know why success is false

File: private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts [hunk 2]
```diff
@@ -1091,7 +1166,7 @@ export class TemplateServiceClient {
       });
     });
 
-    router.post("/:id*?", async (req: Request, res: Response, _next: NextFunction) => {
+    router.post("/:id?", async (req: Request, res: Response, _next: NextFunction) => {
```
Comment (turalf) [inline private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts]: This file is getting bigger. I would consider putting routes to another file. Maybe part of next PRs

Comment (gracez72) [reply private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts]: Ok will do!

File: private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts (hunk 1 above)
Comment (gracez72) [reply private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts]: Fixed, thanks!

File: private-templates-service/adaptivecards-templating-client/typescript/api.ts [hunk 3]
```diff
@@ -845,6 +913,132 @@ export class TemplateApi {
...
     }
+    /**
+     * 
+     * @summary Batch delete template version operation
+     * @param templateId ID of template to update
+     * @param versionList template versions to delete
+     * @param {*} [options] Override http request options.
+     */
+    public batchTemplateDelete (templateId: string, versionList: VersionList, options: any = {}) : Promise<{ response: http.IncomingMessage; body?: any;  }> {
+        const localVarPath = this.basePath + '/template/{templateId}/batch'
```
Comment (turalf) [inline private-templates-service/adaptivecards-templating-client/typescript/api.ts]: batchTemplateDelete and batchTemplateUpdate are almost identical, maybe consider having a private method to be called by those?

Comment (gracez72) [reply private-templates-service/adaptivecards-templating-client/typescript/api.ts]: The files under typescript were autogenerated by the swagger file but I'll keep it in mind for final clean up!

**Pull request merged**